src/overlays/actors/ovl_En_Test3/z_en_test3.c

```diff
@@ -27,113 +203,1114 @@ const ActorInit En_Test3_InitVars = {
+            return ret;
+        }
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3EB8C.s")
+s32 func_80A3ECEC(EnTest3* this, PlayState* play) {
+    return D_80A4169C[this->unk_D78->unk_0](this, play);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3EBFC.s")
+s32 func_80A3ED24(EnTest3* this, PlayState* play) {
+    s32 temp = D_80A416C0[this->unk_D78->unk_0](this, play);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3EC30.s")
+    if (temp != 0) {
+        this->unk_D78 = &this->unk_D78[temp];
+        return func_80A3ECEC(this, play);
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3EC44.s")
+void EnTest3_Init(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    EnTest3* this = THIS;
+    Camera* subCam;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A3ECEC.s")
+    if (D_80A41D24) {
+        Actor_MarkForDeath(&this->player.actor);
+        return;
+    }
+    D_80A41D24 = true;
+    this->player.actor.room = -1;
+    this->player.unk_A86 = -1;
+    this->player.transformation = PLAYER_FORM_HUMAN;
+    this->player.ageProperties = &sAgeProperties;
+    this->player.itemActionParam = PLAYER_AP_NONE;
+    this->player.heldItemId = ITEM_OCARINA;
+    Player_SetModelGroup(&this->player, 3);
+    play->playerInit(&this->player, play, &object_test3_Skel_00F7EC);
+    Effect_Add(play, &this->player.meleeWeaponEffectIndex[0], EFFECT_BLURE2, 0, 0, &sBlureInit);
+    Effect_Add(play, &this->player.meleeWeaponEffectIndex[1], EFFECT_BLURE2, 0, 0, &sBlureInit);
+    Effect_Add(play, &this->player.meleeWeaponEffectIndex[2], EFFECT_TIRE_MARK, 0, 0, &sTireMarkInit);
+    this->player.maskObjectSegment = ZeldaArena_Malloc(0x3800);
+    play->func_18780(&this->player, play);
+    this->unk_D90 = GET_PLAYER(play);
+    this->player.giObjectSegment = this->unk_D90->giObjectSegment;
+    this->player.tatlActor = this->unk_D90->tatlActor;
+    if ((CURRENT_DAY != 3) || (gSaveContext.save.weekEventReg[33] & 8) || !(gSaveContext.save.weekEventReg[51] & 8)) {
+        this->player.currentMask = PLAYER_MASK_KEATON;
+    }
+    this->player.prevMask = this->player.currentMask;
+    if (play->sceneNum == SCENE_SECOM) {
+        this->subCamId = Play_CreateSubCamera(play);
+        subCam = Play_GetCamera(play, this->subCamId);
+        Camera_InitPlayerSettings(subCam, &this->player);
+        Camera_SetFlags(subCam, CAM_STATE_0 | CAM_STATE_6);
+        Play_ChangeCameraStatus(play, this->subCamId, CAM_STATUS_WAIT);
+    }
+    this->player.actor.colChkInfo.cylRadius = 20;
+    this->player.actor.colChkInfo.cylHeight = 60;
+    this->player.actor.colChkInfo.health = 0xFF;
+    if (KAFEI_GET_PARAM_1E0(&this->player.actor) == 0) {
+        func_80A3E7E0(this, func_80A40824);
+    } else {
+        func_80A3E7E0(this, func_80A40678);
+    }
```

Comment:
```suggestion

    if ((CURRENT_DAY != 3) || (gSaveContext.save.weekEventReg[33] & 8) || !(gSaveContext.save.weekEventReg[51] & 8)) {
        this->player.currentMask = PLAYER_MASK_KEATON;
    }
    this->player.prevMask = this->player.currentMask;

    if (play->sceneNum == SCENE_SECOM) {
        this->subCamId = Play_CreateSubCamera(play);
        subCam = Play_GetCamera(play, this->subCamId);
        Camera_InitPlayerSettings(subCam, &this->player);
        Camera_SetFlags(subCam, CAM_STATE_0 | CAM_STATE_6);
        Play_ChangeCameraStatus(play, this->subCamId, CAM_STATUS_WAIT);
    }

    this->player.actor.colChkInfo.cylRadius = 20;
    this->player.actor.colChkInfo.cylHeight = 60;
    this->player.actor.colChkInfo.health = 0xFF;

    if (KAFEI_GET_PARAM_1E0(&this->player.actor) == 0) {
        func_80A3E7E0(this, func_80A40824);
    } else {
        func_80A3E7E0(this, func_80A40678);
    }
```
to make this a bit more readable. Also 0xFF should probably be 255.